process/block/preprocess/transactions.go

```diff
@@ -996,6 +1013,8 @@ func (txs *transactions) createAndProcessMiniBlocksFromMe(
 
 	miniBlocks := txs.getMiniBlockSliceFromMap(mapMiniBlocks)
 
+	txs.gasHandler.AddTotalGasConsumedInSelfShard(totalGasConsumedInSelfShard)
```

Comment:
This will add again the gas consumed in process txs dest me. Maybe we should add in L814: oldTotalGasConsumedInSelfShard := totalGasConsumedInSelfShard and here we should use totalGasConsumedInSelfShard - oldTotalGasConsumedInSelfShard